### PR TITLE
HIVE-26543: Improve TxnHandler, TxnUtils, CompactionTxnHandler logging

### PR DESCRIPTION
**Pull request by abstractdog**

### CODE REVIEW

**Review by achennagiri — COMMENTED** [review]
There are similar improvements that can be done in https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java and https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnUtils.java . Can we include that in this PR as well. The rest of the files in the txn folder seem to be good.

**Review by achennagiri — APPROVED** [review]

**Review by zabetak — COMMENTED** [review]
LGTM, left some minor comments.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 1]
```diff
@@ -716,11 +716,11 @@ private List<Long> openTxns(Connection dbConn, OpenTxnRequest rqst)
 
         if (!targetTxnIdList.isEmpty()) {
           if (targetTxnIdList.size() != rqst.getReplSrcTxnIds().size()) {
-            LOG.warn("target txn id number " + targetTxnIdList.toString() +
-                    " is not matching with source txn id number " + rqst.getReplSrcTxnIds().toString());
+            LOG.warn("target txn id number {} is not matching with source txn id number {}",
+                targetTxnIdList.toString(), rqst.getReplSrcTxnIds().toString());
```
[the remark this answers is not in the document]
Comment (abstractdog) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:720]: ack

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 2]
```diff
@@ -2149,8 +2172,8 @@ public AllocateTableWriteIdsResponse allocateTableWriteIds(AllocateTableWriteIds
           if (srcTxnIds.size() != txnIds.size()) {
             // Idempotent case where txn was already closed but gets allocate write id event.
             // So, just ignore it and return empty list.
-            LOG.info("Idempotent case: Target txn id is missing for source txn id : " + srcTxnIds.toString() +
-                    " and repl policy " + rqst.getReplPolicy());
+            LOG.info("Idempotent case: Target txn id is missing for source txn id : {} and repl policy {}",
+                srcTxnIds.toString(), rqst.getReplPolicy());
```
[the remark this answers is not in the document]
Comment (abstractdog) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2176]: ack

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 3]
```diff
@@ -1196,11 +1199,13 @@ private void updateDatabaseProp(Connection dbConn, String database,
...
       closeStmt(pst);
       if (query == null) {
-        LOG.info("Database property: " + prop + " with value: " + propValue + " already updated for db: " + database);
+        LOG.info("Database property: {} with value: {} already updated for db: {}", prop, propValue, database);
         return;
       }
       pst = sqlGenerator.prepareStmtWithParameters(dbConn, query, Arrays.asList(propValue));
-      LOG.debug("Updating " + prop + " for db: " + database + " <" + query.replaceAll("\\?", "{}") + ">", propValue);
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Updating " + prop + " for db: " + database + " <" + query.replaceAll("\\?", "{}") + ">", propValue);
```
[the remark this answers is not in the document]
Comment (abstractdog) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:1207]: oh, makes sense, we replacing a single char here

**Comment by abstractdog** [comment]
thanks @achennagiri and @zabetak, I addressed your comments in https://github.com/apache/hive/pull/3603/commits/4634139b69b2ee652cc368659242741bef1c6603

**Review by zabetak — APPROVED** [review]

**Pull request merged**